x/staking/types/validator.go

```diff
@@ -9,7 +9,7 @@ import (
 
 	abci "github.com/tendermint/tendermint/abci/types"
 	tmprotocrypto "github.com/tendermint/tendermint/proto/tendermint/crypto"
-	"gopkg.in/yaml.v2"
+	"sigs.k8s.io/yaml"
```

Comment:
sigs.k8s.io/yaml uses `json.Marshal` under the hood, but that doesn't work well with Protobuf anys. What I propose is to use Proto JSON here:

```go
func (v Validator) String() string {
	a, err := codec.ProtoMarshalJSON(&v, nil)
	if err != nil {
		panic(err)
	}

	out, err := yaml.JSONToYAML(a)
	if err != nil {
		panic(err)
	}

	return string(out)
}
```

This should solve the failing test.